deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -227,26 +212,33 @@ class WebxdcViewController: WebViewViewController {
             }
         }
     }
-    
-    private func updateWebxdc(statusId: Int) {
-        let statusUpdates = self.dcContext.getWebxdcStatusUpdates(msgId: messageId, statusUpdateId: statusId)
-        logger.debug("status updates: \(statusUpdates)")
-        webView.evaluateJavaScript("window.__webxdcUpdate(atob(\"\(statusUpdates.toBase64())\"))", completionHandler: nil)
+
+    var lastSerial: Int?
+    private func updateWebxdc() {
+        if let lastSerial = lastSerial {
+            let statusUpdates = dcContext.getWebxdcStatusUpdates(msgId: messageId, lastKnownSerial: lastSerial)
+            if let data: Data = statusUpdates.data(using: .utf8),
+               let array = (try? JSONSerialization.jsonObject(with: data, options: [])) as? [Any],
+               let first = array.first as? [String: Any],
+               let maxSerial = first["max_serial"] as? Int {
+                self.lastSerial = maxSerial
+            }
```

Comment:
I don't really get this strange syntax:
what curly braces terminate which block?
what's with the indentation?

Reply:
the block is only the single line `self.lastSerial = maxSerial` - but that is the default swift formatting, @cyBerta is already very used to that, i was also stumbling upon that, thinking the block also includes the `let`-lines. 

would be better if the `{` is in a single line and on the same indetation as `}` - but swift then complains (and it was not important enough to check how to tweak these linter things ;)